nano/core_test/bootstrap.cpp

```diff
@@ -1345,10 +1334,7 @@ TEST (bootstrap_processor, lazy_pruning_missing_block)
 	// Some blocks cannot be retrieved from pruned node
 	node2->block_processor.flush ();
 	ASSERT_EQ (1, node2->ledger.cache.block_count);
-	ASSERT_FALSE (node2->ledger.block_or_pruned_exists (send1->hash ()));
-	ASSERT_FALSE (node2->ledger.block_or_pruned_exists (send2->hash ()));
-	ASSERT_FALSE (node2->ledger.block_or_pruned_exists (open->hash ()));
-	ASSERT_FALSE (node2->ledger.block_or_pruned_exists (state_open->hash ()));
+	ASSERT_FALSE (nano::test::block_or_pruned_exists (*node2, { send1, send2, open, state_open }));
```

Comment:
The ASSERT_FALSE variants of this aren't equivalent to what they replace. This old test is that none of the blocks exist but the new test is only in any of them don't exist.

It would need a different function similar to std::all_of and std::none_of.  https://en.cppreference.com/w/cpp/algorithm/all_any_none_of